src/main/cluster.ts

```diff
@@ -82,8 +82,7 @@ export class Cluster implements ClusterModel {
   @action
   updateModel(model: ClusterModel) {
     Object.assign(this, model);
-    this.apiUrl = this.getKubeconfig().getCurrentCluster()?.server;
-    this.contextName = this.contextName || this.preferences.clusterName;
+    this.apiUrl = this.getKubeconfig().getCluster(this.getKubeconfig().getContextObject(this.contextName).cluster).server
```

Comment:
Maybe it's better to move `this.getKubeconfig()` into some `const` before this line to avoid parsing kubeconfig twice.

Reply:
Done. Also moved this assign into constructor, since it's not going to change.